GCMtools/utils/interface.py

```diff
@@ -10,27 +10,406 @@
 #        - ...
 # ==============================================================
 
+import numpy as np
+import xarray as xr
+
+from ..core.const import VARNAMES as c
+
+
+class _Chemistry:
+    """
+    Chemistry class used to deal with different kinds of chemical models.
+    """
+
+    def __init__(self):
+        """
+        Constructor for the Chemistry class
+        """
+        self.abunds = xr.Dataset()
+
+    def set_data(self, ds):
+        """
+        Construct the chemistry class
+
+        Parameters
+        ----------
+        ds: DataSet
+            A GCMtools-compatible dataset of a 3D climate simulation.
+            Should not have a time coordinate anymore.
+
+        """
+        self.ds = ds
+
+    def chem_from_poorman(self, temp_key='T', CO=0.55, FeH=0.0):
+        """
+        Calculate equilibrium abundancies with poorman from pRT
+
+        Parameters
+        ----------
+        temp_key: str, optional
+            The key to the temperature field used for the abundancies.
+            Defaults to T.
+        CO: float, optional
+            The C/O ratio. Currently only one global value allowed. Defaults to 0.55.
+        FeH: float, optional
+            The metalicity ratio. Currently only one global value allowed. Defaults to 0.0.
+        """
+        from petitRADTRANS.poor_mans_nonequ_chem import interpol_abundances
+
+        if c['time'] in self.ds.dims:
+            raise ValueError('Dataset should not have a timedimension. Select the timestamp beforehand.')
+
+        p = self.ds[c['Z']].values
```

Comment:
There is currently no flag that a ds is acutally loaded. Is the default error thrown if self.ds does not exist the one we want or do we need to handle this?

Reply:
good point and certainly true here! I will add a handle for that case.